chart/values.schema.json

```
@@ -2069,11 +2258,38 @@
                         "annotations": {
                             "description": "Annotations for the webserver Service.",
                             "type": "object",
-                            "default": {}
+                            "default": {},
+                            "additionalProperties": {
+                                "type": "string"
+                            }
                         },
                         "ports": {
                             "description": "Ports for the webserver Service.",
                             "type": "array",
+                            "items": {
+                                "type": "object",
+                                "additionalProperties": false,
+                                "properties": {
+                                    "name": {
+                                        "type": "string"
+                                    },
+                                    "port": {
+                                        "type": [
+                                            "string",
+                                            "integer"
+                                        ]
+                                    },
+                                    "targetPort": {
+                                        "type": [
+                                            "string",
+                                            "integer"
+                                        ]
+                                    },
+                                    "protocol": {
+                                        "type": "string"
+                                    }
+                                }
+                            },
```

We should be able to use `ServicePort` here (and with flower), no?

```suggestion
                            "items": {
                                "$ref": "https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.22.0/_definitions.json#/definitions/io.k8s.api.core.v1.ServicePort"
                            },
```

---

No. It is not compatible with current default value:
```
                                {
                                    "name": "airflow-ui",
                                    "port": "{{ .Values.ports.airflowUI }}"
                                }
```
In the Kubernetes schema, the `port` field is defined as `integer`, but we use `string`.

---

Ah, good catch.